tools/wptrunner/wptrunner/executors/executorwebdriver.py

```diff
@@ -897,17 +897,28 @@ async def process_bidi_event(method, params):
             # Use protocol loop to run the async cleanup.
             protocol.loop.run_until_complete(protocol.bidi_events.unsubscribe_all())
 
-        extra = {}
-        if (leak_part := getattr(protocol, "leak", None)) and (counters := leak_part.check()):
-            extra["leak_counters"] = counters
-
         # Attempt to clean up any leftover windows, if allowed. This is
         # preferable as it will blame the correct test if something goes wrong
         # closing windows, but if the user wants to see the test results we
         # have to leave the window(s) open.
         if self.cleanup_after_test:
             protocol.testharness.close_old_windows()
 
+        extra = {}
+        if (leak_part := getattr(protocol, "leak", None)) and protocol.testharness.persistent_test_window:
```

Comment:
I feel a little bit weird that leak check needs to have a connection with 'reuse_window'. Maybe what we should do here is: 1. close unrelated windows, 2. run leak check, 3. close the test windows if not going to be reused.

Reply:
Done